### PR TITLE
M2P-191 Fix error issues reported by the Magento coding standard tool in the model folder

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
This PR fixes the error issues reported by the Magento coding standard tool

The following command to run:
vendor/bin/phpcs --standard=Magento2 app/code/Bolt/Boltpay/Model -n

See the reported issues in the following link: https://gist.github.com/johnmccombs1/020cbd9361364493a2a2ec341a08d1b9

Fixes: https://boltpay.atlassian.net/browse/M2P-191

#changelog M2P-191 Fix error issues reported by the Magento coding standard tool in the model folder

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Model/Api/DiscountCodeValidation.php [hunk 1]
```diff
@@ -319,7 +319,7 @@ public function validate()
             $parentQuote->getStore()->setCurrentCurrencyCode($parentQuote->getQuoteCurrencyCode());
 
             // get the coupon code
-            $discount_code = @$request->discount_code ?: @$request->cart->discount_code;
+            $discount_code = $request->discount_code ?? $request->cart->discount_code ?? null;
```
Comment (danac-gs) [inline Model/Api/DiscountCodeValidation.php:322]: This is a PHP7 feature, not backward compatible with PHP5.6.
Should we support merchants running PHP5.6 / Magento2.1?
Officially no, I assume, but do we have any old merchant stores running on PHP5.6?
@matt-thomason

Comment (johnmccombs1) [reply Model/Api/DiscountCodeValidation.php:322]: @danac-gs  @matt-thomason  
The reason I made this change is that Magento 2.1 is already on end-of-life and PHP 5.6 is pretty slow for Magento. So most of the merchants would likely upgrade to Magento 2.3 (or above) or even if they are on Magento 2.1 they would still use PHP7. I've done integration for a lot of merchants and I haven't seen any that used Magento 2.1

I can switch the syntax back to 
            $discount_code = isset($request->discount_code) ? $request->cart->discount_code : null;

But it is more verbose and I think it's now time to specify that our module is only working on PHP7 in composer.json

**Comment by johnmccombs1** [comment]
@njfaries  I've updated code. Can you re-check?

**Comment by njfaries** [comment]
PR looks good, but dropping 5.6 will cause issues with integration test. We are dropping support for it but need to hold off on merging such changes just yet. cc @matt-thomason

**Comment by njfaries** [comment]
@johnmccombs1 7.0 compatibility is no longer an issue, reopening this to get it merged in for 2.14

**Pull request merged**